### PR TITLE
HUD: Reject and warn on invalid stat types

### PR DESCRIPTION
**Pull request by SmallJoker**
Trivial fix for #11539

## To do

This PR is Ready for Review.

## How to test

1. Check whether  HUD element changes still work, for example using `/hudfonts` in devtest
2. Modify `hud_change` calls to change `TEXT` or `sTyLe` and observe unique logs

### CODE REVIEW

**Comment by AFCMS** [comment]
Why `WIP` label and "Ready for review" ?

File: src/network/clientpackethandler.cpp [hunk 1]
```diff
@@ -1119,17 +1119,29 @@ void Client::handleCommand_HudChange(NetworkPacket* pkt)
...
+		case HUD_STAT_WORLD_POS:
+			*pkt >> v3fdata;
+			break;
+		case HUD_STAT_SIZE:
+			*pkt >> v2s32data;
+			break;
+		default:
+			*pkt >> intdata;
+			return;
+	}
```
Comment (sfan5) [inline src/network/clientpackethandler.cpp:1144]: This is broken, there's a `return;` at the end.

File: src/script/common/c_internal.h [hunk 2]
```diff
@@ -134,5 +137,4 @@ DeprecatedHandlingMode get_deprecated_handling_mode();
  * @param message The deprecation method
  * @param stack_depth How far on the stack to the first user function (ie: not builtin or core)
  */
-void log_deprecated(lua_State *L, const std::string &message,
-	int stack_depth=1);
+void log_deprecated(lua_State *L, std::string message, int stack_dept = 1);
```
Comment (sfan5) [inline src/script/common/c_internal.h:140]:
```suggestion
void log_deprecated(lua_State *L, std::string message, int stack_depth = 1);
```

File: src/network/clientpackethandler.cpp [hunk 3]
```diff
@@ -1119,17 +1119,29 @@ void Client::handleCommand_HudChange(NetworkPacket* pkt)
...
+			break;
+		case HUD_STAT_WORLD_POS:
+			*pkt >> v3fdata;
+			break;
+		case HUD_STAT_SIZE:
+			*pkt >> v2s32data;
+			break;
+		default:
+			*pkt >> intdata;
+			return;
```
Comment (nerzhul) [inline src/network/clientpackethandler.cpp:1143]: replace with break, else you discard data

**Review by nerzhul — CHANGES_REQUESTED** [review]

**Comment by sfan5** [comment]
I don't see anything logged when executing `executed core.get_player_by_name("singleplayer"):hud_change(3, "WRONG", 1)` though the hud packet doesn't go out.

**Review by sfan5 — APPROVED** [review]

**Pull request merged**